python/dgl/contrib/dis_sampling/receiver.py

```diff
@@ -0,0 +1,74 @@
+# This code implements a distributed dgl-subgraph receiver.
+from concurrent import futures
+
+import mxnet as mx
+import numpy as np
+
+import time
+from queue import Queue
+
+import grpc
+import sender
+import sampler_pb2
+import sampler_pb2_grpc
+
+_MAX_MESSAGE_LENGTH = 1024 ** 3  # Gigabyte
+
+class Receiver(sampler_pb2_grpc.SamplerServicer):
+    """ Reciever is called by trainer and it uses back-end thread
+        to receive sub-graph samples from remote sampler.
+    """
+    def __init__(self, port, buffer_size):
+    	self.port = port;
+    	self.queue = Queue(maxsize = buffer_size)
+
+    def SendSubGraph(self, request, context):
+        """ Recv sub-graph, re-construct it, and 
+            put it into the queue.
+
+        Args:
+          request: grpc request
+          context: grpc context
+        """
+        ver_id = sender.deserilize(request.vertices_id, np.int64)
+        layer = sender.deserilize(request.layer, np.int64)
+        csr_data = sender.deserilize(request.csr_data, np.int64)
+        csr_indices = sender.deserilize(request.csr_indices, np.int64)
+        csr_indptr = sender.deserilize(request.csr_indptr, np.int64)
+        csr_shape_0 = request.csr_shape_0
+        csr_shape_1 = request.csr_shape_1
+
+        csr_graph = mx.nd.sparse.csr_matrix((
+            csr_data, 
+            csr_indices, 
+            csr_indptr), 
+          shape=(csr_shape_0, csr_shape_1))
```

Comment:
Could we use `dgl.graph_index.GraphIndex` object instead of mx csr data? It's better not depend on mx. The CSR data structure would be enabled once #342 is done.

Reply:
On a second thought, this requires the graph index class to have `ConvertToBytes` and `ConvertFromBytes` functions, which should be doable.